src/v/kafka/protocol/request_reader.h

```diff
@@ -170,6 +170,21 @@ class request_reader {
         return do_read_array(len - 1, std::forward<ElementParser>(parser));
     }
 
+    void consume_tags() {
+        // Only relevent when reading flex requests
+        auto len = read_unsigned_varint();
+        if (len > 0) {
+            for (; (len - 1) > 0; --len) {
```

Comment:
This looks suspicious. Length of tags is len - 1 but this loop isn't consuming the final element. `>=` seems more correct, or will swap for while loop